couchdb/server/state.py

```diff
@@ -2,11 +2,37 @@
 #
 import logging
 
-__all__ = ('add_fun', 'add_lib')
+__all__ = ('add_fun', 'add_lib', 'reset')
 
 log = logging.getLogger(__name__)
 
 
+def reset(server, config=None):
+    """Resets query server state.
+
+    :command: reset
+
+    :param server: Query server instance.
+    :type server: :class:`~couchdb.server.BaseQueryServer`
+
+    :param config: Optional dict argument to set up query config.
+    :type config: dict
+
+    :return: True
+    :rtype: bool
+    """
+    log.debug('Reset server state')
+    del server.state['functions'][:]
+    del server.state['functions_src'][:]
+    server.state['query_config'].clear()
+    if config is not None:
+        log.debug('Set new query config:\n%s', config)
+        server.state['query_config'].update(config)
+    if server.version >= (1, 1, 0):
+        server.state['view_lib'] = ''
```

Comment:
No reason since it makes yet another way to set the state value. Better have it consistent. Those property is related to the view_lib query command. May be it have to be a function, not a property. Because consistency. Never actually have any need to use it (: